svg/svg.go

```diff
@@ -92,7 +94,45 @@ func NewWindowControls(r float64, x, y float64) *etree.Element {
 		circle.CreateAttr("fill", color)
 		bar.AddChild(circle)
 	}
-	return bar
+	controlsWidth := float64(len(bar.ChildElements()))*float64(x) + float64(r)*2
+	return bar, controlsWidth
+}
+
+// NewWindowTitle returns a title element with the given text.
+func NewWindowTitle(positions []float64, position string, fs float64, ff, text string, s *chroma.Style) (*etree.Element, error) {
+	if text == "" || text == "-" {
```

Comment:
If you pass a Title object as I suggested in other comment, you could move this to a validator function you will add to Title struct.

It would simplify the code here and would move the logic out of this method